addons/hr_holidays/tests/test_leave_requests.py

```diff
@@ -227,7 +227,9 @@ def test_department_leave(self):
     def test_allocation_request(self):
         """ Create an allocation request """
         # employee should be set to current user
-        allocation_form = Form(self.env['hr.leave.allocation'].with_user(self.user_employee))
+        # the default holiday_status_id is not set but needed for security rules,
+        # add it to the context
+        allocation_form = Form(self.env['hr.leave.allocation'].with_user(self.user_employee).with_context(default_holiday_status_id=self.holidays_type_2.id))
```

Comment:
Without the context and without a default value (there is a function that get a default, but there is none set for the test), I get an access error because of the security rule stating `holiday_status_id.company_id in ...`. That field is required, but it's not yet set when creating a new record if there are no default values.
I updated the test here. Another solution was updating the security rule.

Reply:
This shows the side effect of your changes, instead of "fixing" it, I would be checking what is that I didn't foresee

Reply:
Actually the existing code was working by accident.  I removed the nonsensical code that was breaking.